api/src/main/java/org/pmiops/workbench/cohorts/CohortCloningService.java

```diff
@@ -26,15 +26,16 @@
 
   @Transactional
   public Cohort cloneCohortAndReviews(Cohort fromCohort, Workspace targetWorkspace) {
-    Cohort toCohort =
+    final Cohort duplicatedCohort =
         cohortFactory.duplicateCohort(
             fromCohort.getName(), targetWorkspace.getCreator(), targetWorkspace, fromCohort);
-    cohortDao.save(toCohort);
+    final Cohort toCohort = cohortDao.save(duplicatedCohort);
```

Comment:
nit: maybe we can pass the call of `duplicateCohort` directly into this save call? We shouldn't be using `duplicatedCohort` after this point right?

same comment for the save below.

Reply:
I rarely find nested calls to be more readable. In this case, with multiple arguments that themselves already call methods, I don't think that would be a win. 

The other reason is for step-ability, although IJ is getting really good at that and making it less of an issue.

Reply:
I don't think its more readable either. I think the benefit is that it decreases the likelihood of misuse. I'm guessing we should only be using `toCohort` and any calls made on `duplicatedCohort` can be faulty.

Reply:
The usual solution to avoiding variable reuse is keeping scopes small. There are plenty of ways to misuse any number of variables. I don't love the convention of using the same structure for both unsaved and saved models. Ultimately, I think the confusion you get from nesting easily outweighs the potential to use the wrong variable. They're both in scope only a short time, and I felt that calling out that the return value is important could help with avoiding the situation that caused this bug.

Reply:
OK, I just now realized that the [save](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#save-S-) method can mutate its parameter in arbitrary ways. So yes, that variable is more of a live wire than I gave it credit for. In the particular case of APIs that feel the need to alter their arguments in unpredictable ways, avoiding keeping that parameter around is definitely better.